src/main/java/org/wyvie/chehov/bot/commands/CommandProcessor.java

```diff
@@ -65,9 +69,20 @@ public void processKarma(Message message) {
         else if (emojiHelper.isThumbsDown(messageText))
             messageText = "-";
 
-        getCommandHandler("karma" + messageText).ifPresent(handler ->
-                handler.handle(message, "")
-        );
+        getCommandHandler("karma" + messageText).ifPresent(handler -> {
+            if (canProcessCommand(message, handler)) {
+                handler.handle(message, "");
+            }
+        });
+    }
+
+    private boolean canProcessCommand(Message message, CommandHandler handler) {
+        boolean canProcess = false;
+        if (!(handler.getClass().isAnnotationPresent(PublicChatOnlyCommand.class) &&
+                message.chat().type() == Chat.Type.Private)) {
```

Comment:
So is it public chat only or private chat only command?

Reply:
If command is annotated with "PublicChatOnly" and chat is of type "Private" we won't process (ignore) otherwise will process.